### PR TITLE
Add GetRandomWordAPI

### PR DESCRIPTION
**Pull request by MarcRaeder**
Add GetRandomWordAPI

### CODE REVIEW

File: GetRandomWordAPI/GetRandomWordAPI/Repository/GetRandomWordRepository.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        this.DataService = dataService;
+    }
+    
+    public List<string> GetAllWords()
+    {
+        return DataService.wordlist;
+    }
+    
+    public IEnumerable<string> GetWord(int number)
```
Comment (stefanluth) [inline GetRandomWordAPI/GetRandomWordAPI/Repository/GetRandomWordRepository.cs]:
```suggestion
    public IEnumerable<string> GetWords(int number)
```

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    [ApiController]
+    public class GetRandomWordController : ControllerBase
+    {
+        private GetRandomWordService GetRandomWordService;
+
+        public GetRandomWordController(GetRandomWordService getRandomWordService)
+        {
+            this.GetRandomWordService = getRandomWordService;
+        }
+        // GET: api/GetRandomWord
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs]:
```suggestion

```

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        [HttpGet("/all")]
+        public List<string> GetAllWords()
+        {
+            return GetRandomWordService.GetAllWords();
+        }
+
+        // GET: api/GetRandomWord/5
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs]:
```suggestion
```

File: GetRandomWordAPI/GetRandomWordAPI/Program.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+var  MyAllowSpecificOrigins = "_myAllowSpecificOrigins";
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddCors(policyBuilder =>
+    policyBuilder.AddDefaultPolicy(policy =>
+        policy.WithOrigins("https://localhost:7224").AllowAnyHeader().AllowAnyHeader())
+);
+
+// Add services to the container.
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Program.cs]: Remove comments ;)

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using System.Threading.Tasks;
+using GetRandomWordAPI.Service;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GetRandomWordAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class GetRandomWordController : ControllerBase
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs]:
```suggestion
    public class WordController : ControllerBase

```

File: GetRandomWordAPI/GetRandomWordAPI/Repository/GetRandomWordRepository.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+using System.Security.Cryptography;
+using GetRandomWordAPI.Service;
+
+namespace GetRandomWordAPI.Repository;
+
+public class GetRandomWordRepository
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Repository/GetRandomWordRepository.cs]:
```suggestion
public class WordRepository
```

File: GetRandomWordAPI/GetRandomWordAPI/Service/GetRandomWordService.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
+using GetRandomWordAPI.Repository;
+
+namespace GetRandomWordAPI.Service;
+
+public class GetRandomWordService
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Service/GetRandomWordService.cs]:
```suggestion
public class WordService
```

Comment (DevBastiS) [reply GetRandomWordAPI/GetRandomWordAPI/Service/GetRandomWordService.cs]: We are naming this kind of services/repos/controllers after the entity it is working with. In this case it is a "word".

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        // GET: api/GetRandomWord/5
+        [HttpGet("/word")]
+        public IEnumerable<string> GetWord(int number)
+        {
+            return GetRandomWordService.GetWord(number);
+        }
+        
+    }
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs:28]: I am missing a "GetRandomWord" function here.

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        // GET: api/GetRandomWord
+        [HttpGet("/all")]
+        public List<string> GetAllWords()
+        {
+            return GetRandomWordService.GetAllWords();
+        }
+
+        // GET: api/GetRandomWord/5
+        [HttpGet("/word")]
+        public IEnumerable<string> GetWord(int number)
```
Comment (DevBastiS) [inline GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs]:
```suggestion
        public IEnumerable<string> GetWord(int amount)

```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs (hunk 8 above)
Comment (MarcRaeder) [reply GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs:28]: with GetWord function if you don't give a number/amount you get a random word

File: GetRandomWordAPI/GetRandomWordAPI/GetRandomWordAPI.csproj [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="6.0.10" />
+        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <Folder Include="Controllers" />
+    </ItemGroup>
```
Comment (stefanluth) [inline GetRandomWordAPI/GetRandomWordAPI/GetRandomWordAPI.csproj]:
```suggestion
    <ItemGroup>
        <Folder Include="Controllers" />
    </ItemGroup>
```

**Review by stefanluth — CHANGES_REQUESTED** [review]
Add EOFs :)

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs [hunk 11]
```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GetRandomWordAPI.Service;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
```
Comment (stefanluth) [inline GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs]: Many unused imports here and in the other files :)

File: GetRandomWordAPI/GetRandomWordAPI/Service/GetRandomWordService.cs [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+public class WordService
+{
+    private WordRepository WordRepository;
+
+    public WordService(WordRepository wordRepository)
+    {
+        this.WordRepository = wordRepository;
+    }
+    public List<string> GetAllWords()
```
Comment (stefanluth) [inline GetRandomWordAPI/GetRandomWordAPI/Service/GetRandomWordService.cs]:
```suggestion
    }

    public List<string> GetAllWords()
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs (hunk 8 above)
Comment (DevBastiS) [reply GetRandomWordAPI/GetRandomWordAPI/Controllers/GetRandomWordController.cs:28]: But you get "RandomWords" not just "Words" it ist just simply the naming that throwing me off here ;)

**Review by DevBastiS — APPROVED** [review]

**Review by stefanluth — APPROVED** [review]

**Pull request merged**